index.html

```diff
@@ -395,6 +404,60 @@
     },
   });
 
+    // Feed the datalist autocomplete values:
+
+    searchSuggestions.innerHTML = graph.nodes().map(node=>"<option value=\"".concat(graph.getNodeAttribute(node, "label"), "\"></option>")).join("\n");
+    // Actions:
+
+    function setSearchQuery(query) {
+        state.searchQuery = query;
+        if (searchInput.value !== query)
+            searchInput.value = query;
+
+        if (query) {
+            const lcQuery = query.toLowerCase();
+            const suggestions = graph.nodes().map(n=>({
+                id: n,
+                label: graph.getNodeAttribute(n, "label")
+            })).filter(({label})=>label.toLowerCase().includes(lcQuery));
+            // If we have a single perfect match, them we remove the suggestions, and
+            // we consider the user has selected a node through the datalist
+            // autocomplete:
+
+            if (suggestions.length === 1 && suggestions[0].label === query) {
+                state.selectedNode = suggestions[0].id;

```

Comment:
Can you merge `state.selectedNode` and `state.defaultNode`?  I think they achieve the same goal here.